packages/fast-components-react-base/src/image/image.tsx

```diff
@@ -5,101 +5,65 @@ import Foundation, { HandledProps } from "../foundation";
 
     /**
      * Renders the component
      */
     public render(): React.ReactElement<HTMLImageElement | HTMLPictureElement> {
-        if (!this.props.src && !this.props.vp1) {
-            return;
-        }
+        let className: string = get(this.props, "managedClasses.image");
 
-        if (this.props.src) {
+        if (!this.props.children) {
             return (
                 <img
                     {...this.unhandledProps()}
-                    className={super.generateClassNames(get(this.props, "managedClasses.image"))}
+                    className={super.generateClassNames(className)}
                     alt={this.props.alt}
                     sizes={this.props.sizes ? this.props.sizes : null}
                     src={this.props.src}
                     srcSet={this.props.srcSet ? this.props.srcSet : null}
                 />
             );
         } else {
+            className = `${className} ${get(this.props, "managedClasses.image__picture")}`;
+
             return (
                 <picture
                     {...this.unhandledProps()}
-                    // tslint:disable-next-line:max-line-length
-                    className={super.generateClassNames(`${get(this.props, "managedClasses.image")} ${get(this.props, "managedClasses.image__picture")}`)}
+                    className={super.generateClassNames(className)}
                 >
-                    {this.generateSourceElement(6)}
-                    {this.generateSourceElement(5)}
-                    {this.generateSourceElement(4)}
-                    {this.generateSourceElement(3)}
-                    {this.generateSourceElement(2)}
-                    {this.generateSourceElement(1)}
+                    {this.renderChildrenBySlot(ImageSlot.source)}
                     <img
-                        src={this.props.vp1}
+                        src={this.props.src}
                         alt={this.props.alt}
-                        className={super.generateClassNames(get(this.props, "managedClasses.image_img"))}
+                        className={get(this.props, "managedClasses.image_img")}
                     />
                 </picture>
             );
         }
     }
 
     /**
-     * Generate a source element
+     * Render children by slot
      */
-    private generateSourceElement(breakpointsId: number): any {
-        const srcset: string = this.props[`vp${breakpointsId}`];
-        const vp: number = breakpoints[`vp${breakpointsId}`];
-        const minWidth: number = vp;
-
-        if (!srcset) {
-            return null;
-        }
-
-        const minWidthString: string = minWidth === 0 ? "0" : `${minWidth}px`;
-
-        return <source srcSet={srcset} media={`(min-width: ${minWidthString})`} />;
+    private renderChildrenBySlot(slot: ImageSlot): Array<React.ReactElement<HTMLSourceElement>> {
+        return React.Children.map(this.props.children, (child: React.ReactElement<HTMLSourceElement>, index: number) => {
```

Comment:
Your callback doesn't have a return type - not sure why it is passing tslint but it shouldn't be

Reply:
Added - had to update formatting as well due to line length.